### ISSUE TITLE
Generate static HTML files with the same look as the Talawa documentation website

### ISSUE THREAD

**Issue by DMills27** [PalisadoesFoundation/talawa-docs#199]
Using the guide found [here](https://v1.docusaurus.io/docs/en/custom-pages), try to mimic the look and feel of the docusaurus pages on found throughout the site with a static HTML file of your creation. The purpose of the issue is to essentially create a template from which code can be placed inside and the page rendered will be consistent with the rest of the site in terms of styling.

**Parent Issue**
- https://github.com/PalisadoesFoundation/talawa-docs/issues/191

**Comment by madhhuurrii** [comment]
Hi, I am interested in contributing to this issue. I would like to know if we have to create html file pages for each page one by one or else

**Comment by DMills27** [comment]
No, you just need to create a template essentially that matches the header and footer of the site and accommodates code being placed in the "middle". What this means is that for every successful pull request, items/links would be added to this template. The goal is to essentially create a page whose HTML won't break when code is added to its body. Think of creating a Table fo contents that will be continuous updated and contain numerous sub-links etc. You can use [this article](https://jsvxc.com/html-table-of-contents-generator/) for a means to go about doing this.  If anything is not clear here please don't hesitate to ask for clarification.

**Comment by Suyash-Purwar** [comment]
Hey, I am interested in contributing to this issue. Can it be assigned to me?

**Comment by Suyash-Purwar** [comment]
I'll reach out to the maintainers you mentioned very soon and will start work after thier guidance.

**Comment by garvit3835** [comment]
Hi, can I work on this issue?

**Comment by palisadoes** [comment]
Please contact @tasneemkoushar on slack. Tasneem can you provide any advice here?

**Comment by tasneemkoushar** [comment]
Hi @garvit3835, Kindly contact me for any queries on slack or here as well. 
https://docs.talawa.io/docs/
Currently, this is our documentation website. Go through the website, and create an HTML template with the same look.

**Comment by garvit3835** [comment]
Hi @tasneemkoushar, so all I have to do is to create a static 'docs' page in the pages section and copy the content and styling of https://docs.talawa.io/docs/ this website. Right?

**Comment by tasneemkoushar** [comment]
Yeah, you can begin with that.
What we are thinking to achieve is to convert the look-for pages similar to:-
https://docs.talawa.io/talawa/views_base_view/BaseView-class.html
to that of our 'docs' page.

If we have a template page, we can write a script to put documentation content and code inside that template and the page rendered will be consistent with the rest of the site in terms of styling.

**Comment by garvit3835** [comment]
@tasneemkoushar I just wanted to confirm something, I have to create .md files? And can you tell me where I can find CSS for the docs or do I have to create the CSS from scratch?

**Comment by itsgurmannatsohal** [comment]
Hello. I am working on issue #200 and needed a rough idea about the static pages that you're creating @garvit3835

**Comment by garvit3835** [comment]
Hi @itsgurmannatsohal, I am creating a markdown page having template similar to the docs page. Most of it is done, just stuck on the sidebar. Still trying to copy the sidebar of the docs page to the template.

**Comment by garvit3835** [comment]
Please let me know if you know how to do it.

**Comment by itsgurmannatsohal** [comment]
okay sure

**Comment by garvit3835** [comment]
@tasneemkoushar, I am having a problem. In order to make a static .html page template of the docs, I would need the CSS file used in the docs and I am not able to find it. Without it, I will not be able to copy the exact styling. Can you tell where is it located?

**Comment by aanya963** [comment]
Hey! @tasneemkoushar  
I'm interested in this issue, Can you assign this to me?

**Comment by ishitakapoor26** [comment]
Hey! @garvit3835 can we collaborate on this issue?

**Comment by palisadoes** [comment]
@garvit3835

1. We have the same issue with Talawa-API and Talawa Documentation.
1.  Which one are you working on so that we can have someone else work on the other?

**Comment by Yass32** [comment]
Hello @tasneemkoushar, I would like to complete this task. I have to mimic the documentation page using HTML, but what about the CSS?

**Comment by david-ibemma** [comment]
hello @palisadoes I am interested in working on this issue. How do I go about it. Efforts to reach out to mentors proved abortive.

**Comment by sakshi4chapke** [comment]
hello @palisadoes I am interested in working on this issue. How do I go about it.

**Comment by tasneemkoushar** [comment]
@garvit3835 @itsgurmannatsohal We require an HTML template, not a .md template that should look similar to pages like [this](https://docs.talawa.io/docs/applications/overview).

**Comment by tasneemkoushar** [comment]
> @tasneemkoushar, I am having a problem. In order to make a static .html page template of the docs, I would need the CSS file used in the docs and I am not able to find it. Without it, I will not be able to copy the exact styling. Can you tell where is it located?

You can take inspiration from [here](https://github.com/PalisadoesFoundation/talawa-docs/tree/develop/src/css). I would suggest writing your own from the scratch. It will be simpler and easier.

**Comment by KASHISH17RAMANI** [comment]
hello everyone i am interested to contribute to this issue . i have installed docasuraus now please guide me how should i go about it

**Comment by Yass32** [comment]
> @garvit3835
> 
> 1. We have the same issue with Talawa-API and Talawa Documentation.
> 2. Which one are you working on so that we can have someone else work on the other?

@palisadoes can I work on the other

**Comment by KASHISH17RAMANI** [comment]
please allow me to contribute in this issue

**Comment by tasneemkoushar** [comment]
@garvit3835 let me know how you're doing

**Comment by singhavs** [comment]
Hello @tasneemkoushar can I get started with this issue?

**Comment by Biki-das** [comment]
@DMills27 rather than adjusting the styles how about we go through populating the docs? we have a plugin called 
[plugin-remote-content](https://www.npmjs.com/package/docusaurus-plugin-remote-content) which is built for the very purpose of fetching docs from other repositories.

My approach is to write a script interacting with the Github api and fetching all the talawa docs from the repository and populate them in our current docs.

Would like to know your thoughts on the same?

**Comment by Biki-das** [comment]
@garvit3835 its more than a week now any feedback on your work,i am digging  on this issue too, we can talk and analyse about this problem

@DMills27  @tasneemkoushar 
Below is the Proof of concept for the current home page of talawa mobile api page, and i have extracted the docusaurus default styles and injected it into a external stylesheet and used it to generate the current html page by pasting the current homepage of the talawa mobile api? am i going in the right direction, now i just need to make the body dynamic such that the content does not break right?


https://user-images.githubusercontent.com/72331432/225523569-4b52c7cf-e63c-482c-857a-b07e9467fdff.mp4

**Comment by suhana832** [comment]
hello @DMills27 ! 
can you assign this issue to me ?

**Comment by Merveille-Louange** [comment]
Hello @garvit3835, given that I am now working on issue 200 which depends on this current issue, we could synchronize to have this fixed so I could proceed with issue 200.

**Comment by Biki-das** [comment]
@palisadoes seems like the other assigned mentors are busy, could you possibly give an update to this one?

**Comment by itstasneem** [comment]
@garvit3835 is there any progress on your side on this issue or should I assign this to someone else?

**Comment by palisadoes** [comment]
Unassigning due to inactivity and no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by Merveille-Louange** [comment]
Hello Peter Harrison,
The issue with this task is that I see it as dependent on a previous task,
issue 119, which unfortunately hasn't been completed yet. I have even
requested to work with the individual who was assigned the task 119 but
haven't had any luck. So I am kinda short in ideas here, but I would really
like to complete this task.

Best regards,
Louange

**Comment by palisadoes** [comment]
PR #119 has been closed and merged last year. Do you have a typo?

- https://github.com/PalisadoesFoundation/talawa-docs/pull/119

**Comment by Biki-das** [comment]
@palisadoes i would like to be assigned if there is no feedback by the current assigned person in the threshold time

**Comment by Merveille-Louange** [comment]
sorry i meant issue 199

**Comment by palisadoes** [comment]
- This is issue 199 #199 
- Which is the dependent one?

**Comment by Merveille-Louange** [comment]
issue 200 is dependent on 199

**Comment by palisadoes** [comment]
@Merveille-Louange 

I think there is come confusion. I think you mean that 
- issue199 #199 
- is dependent on issue 200 #200 

You are assigned issue #200, and based on the miscommunication I also assigned you this issue.

1. Do you want to work on this issue, and are you capable of completing it?
1. I'd prefer not to have any more bottlenecks and have you work on both. If not we can assign this one to someone else.

**Comment by Merveille-Louange** [comment]
Thank you, I will like to have a chance to actually work on both issues

**Comment by palisadoes** [comment]
@Merveille-Louange Both have been assigned to you. Please try to resolve this quickly. The delay was unexpected.

**Comment by Merveille-Louange** [comment]
Thank you.

**Comment by Merveille-Louange** [comment]
Hello @palisadoes, can you please reassign this issue?
there are a lot of things I am still trying to figure out in the project and I am not sure how long before I can get this issue resolved

**Comment by Biki-das** [comment]
@palisadoes could you assign me this one seems like the past assign has excluded them

**Comment by Biki-das** [comment]
@Merveille-Louange whats your status about this issue? could you list down what you figured out till now?

**Comment by tasneemkoushar** [comment]
@Biki-das assigned it to you, you can now start working on it

**Comment by Biki-das** [comment]
> @Biki-das assigned it to you, you can now start working on it

thanks @tasneemkoushar!

**Comment by palisadoes** [comment]
Unassigning due to inactivity and no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

**Comment by Biki-das** [comment]
> Unassigning due to inactivity and no submitted PR. Others need to get a chance. Please let us know whether you want to be reassigned before someone else requests assignment or select another issue.

you can assign it to others, i am still thinking of a better solution
